apps/external_thread_pool_demo/demo_app.cc

```diff
@@ -0,0 +1,24 @@
+/**
+ * DAIL in Alibaba Group
+ *
+ */
+
+#include "demo_app.hh"
+#include "my_thread_pool.hh"
+
+int demo_app::run(int ac, char **av, std::function<void()> &&func) {
+    return _app.run_deprecated(ac, av, [fn = std::move(func)] () mutable {
+        return micro_system::configure().then(
+            std::move(fn)
```

Comment:
missing call operator?
(like `std::move(fn)()`)

Reply:
The usage of seastar/src/core/app-template.cc(line 208) is borrowed here.

Reply:
ah ok, I missed that it was passed as paramter to `.then()`.